tasks/kernel_matrix_testing/README.md

```diff
@@ -1,425 +1,564 @@
+```bash
+dda inv -e kmt.launch-stack --stack=my-first-stack
+```
+> Since this is a local VM, you may be prompted for your password to run some commands with `sudo`.
 
-In all cases, you will be able to input a key name for AWS, in case the keypair you stored in AWS has a different name than the name present in the key file.
+**4. Check Status**
 
-### Create stack
+You can view the running VMs and their IP addresses.
 
 ```bash
-dda inv -e kmt.create-stack --stack=demo-stack
+dda inv -e kmt.status --stack=my-first-stack
 ```
 
-### Configure stack
-
-We will configure the stack to launch
+**5. Connect to the VM**
 
-- Remote x86_64 machine with ubuntu-jammy, ubuntu-focal VMs.
-- Remote arm64 machine with amazon linux 2 kernel 4.14, amazon linux 2 kernel 5.10 VMs.
+The easiest way to connect is by generating an SSH configuration file.
 
 ```bash
-dda inv -e kmt.gen-config --vms=x86-jammy-distro,x86-focal-distro,arm64-amazon4.14-distro,arm64-amazon5.10-distro --stack=demo-stack
+# Generate the SSH config
+dda inv -e kmt.ssh-config --stack=my-first-stack > ~/.ssh/kmt_ssh_config
+
+# Add `Include ~/.ssh/kmt_ssh_config` to your main ~/.ssh/config file.
+# Then, you can connect with:
+ssh kmt-my-first-stack-local-ubuntu_22.04
 ```
 
-We can also configure the stack to launch VMs locally, i.e. on developer's machine. The VMs will have the same architecture as the local host environment.
-- Local amazon linux 5.10 VM.
-- Local fedora 38 VM.
-- Local debian 9 VM.
+**6. Clean Up**
+
+When you are finished, destroy the stack to tear down all associated resources.
 
 ```bash
-dda inv -e kmt.gen-config --vms=amzn5.10-local-distro,fedora38-local-distro,debian9-local-distro --stack=demo-stack
+dda inv -e kmt.destroy-stack --stack=my-first-stack
 ```
 
-Refer to [Configuring the stack](#configuring-the-stack) for more details on this command.
+### Quick Start for Remote VMs
 
+This guide shows a minimal workflow to get a single remote VM running in AWS.
 
-#### Configuring stack from a failed CI pipeline
+**1. Initialize KMT & Configure SSH Key** (only needs to be done once)
 
-A common use case is to configure the stack to replicate the jobs that failed in a CI pipeline. This can be easily done by using the `--from-ci-pipeline` flag, using the ID of the pipeline that failed in Gitlab as the argument:
+To create EC2 instances for remote VMs, KMT needs an SSH key to provision them. The `kmt.init` command will guide you through an interactive wizard to set up a default SSH key, so you don't have to provide it for every command. For a detailed explanation of the supported methods, see the [`kmt.config-ssh-key`](#kmtconfig-ssh-key) command documentation.
 
 ```bash
-dda inv -e kmt.gen-config --from-ci-pipeline=<pipeline-id> --use-local-if-possible --stack=demo-stack
+# Using --remote-setup-only is sufficient if you don't plan to use local VMs
+dda inv -e kmt.init --remote-setup-only
```

Comment:
I would call out that using local VMs is highly preferred for day to day use, since remote VMs launch metal instances which can be costly.

Remote VMs are recommended if a developer is running tests against a large number of VMs concurrently.